src/managedwriter/writer_client.ts

```diff
@@ -102,6 +111,15 @@ export class WriterClient {
     return this._open;
   }
 
+  // Enables StreamConnections to automatically retry failed appends.
+  //
+  // Enabling retries is best suited for cases where users want to achieve at-least-once
+  // append semantics. Use of automatic retries may complicate patterns where the user
+  // is designing for exactly-once append semantics.
+  enableWriteRetries(enable: boolean) {
```

Comment:
I would remove this and enable it by default, the customer could disable it by setting the maxNumRetry to be 0.
_retrySettings can also be confusing because you don't support all request level retries, you only support resend the request when the connection is broken. Maybe cal the member of maxRetryAttemptsOnFailedConnection? It is a mouthful, not sure if there is better name for it.

Reply:
I'm afraid of leaving the write retries as ON by default because it is not recommended on a write-once scenario, unless writes are send in sequence and synchronously. Any thoughts @shollyman ?

Reply:
But do you need to allow users to be able to configure retry times?